magento/pwa-studio
# PWA-1441 [feature]: Switching stores results in error page if slug changes between stores

## Description

When switching store views (frequently languages), if the URL slug isn't the same for the new store view, it will cause a 404. This after plugin will check what the URL is supposed to be for the current store view, based on the old URL, and return a redirect if found.

Related PR for PWA extension https://github.com/magento-commerce/magento2-pwa/pull/10

## Related Issue







Closes #[PWA-1441](https://jira.corp.magento.com/browse/PWA-1441).

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps

1. Configure 3 store views: English, French, Spanish
2. In admin, change the URL for a category in the French and Spanish store views (do not check the "create redirect" checkbox)
3. In admin, disable other category in the Spanish store view
4. Visit the English PWA Studio store front for that category
5. Change to French store view. Should redirect to new URL
6. Change to Spanish store view. Should redirect to new URL
7. Visit the English PWA Studio store front for other category
8. Change to Spanish store view. Should redirect to homepage without errors
9. Return to English store view. Should redirect to new URL
10. Verify Bread crumbs loads with correct name.
10. Repeat steps 2-10 for a product
11. Search for products on English PWA Studio store switch to French store view. Should redirect to search result page correctly
12. Verify product, category switching when url suffix is enabled in backend.
12. Repeat verification steps with enabled/disabled store code in url
13. Run cypress tests

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
